### PR TITLE
little hack to make elasticsearch search OK on indexes WITHOUT any "metadata" field

### PR DESCRIPTION
**Pull request by RodolpheCalvet**
**Description:**
This is for enabling the whole thing for people (like me) who just **don't have "metadata" field in their index**... these extra things I put them at the same level as content in the doc, so using your tools was asking me to add a useless metadata field... I prefered to share this.
I Tested it, looks ok! The search returns the expected doc without any metadata part

**Issue:**
Blocking the search if docs DONT have the "metadata" field.

**Dependencies:**
no change

### CODE REVIEW

**Comment by baskaryan** [comment]
cc @joemcelroy

**Comment by efriis** [comment]
Also if you could get lint passing that would be great! @RodolpheCalvet

File: libs/langchain/langchain/vectorstores/elasticsearch.py [hunk 1]
```diff
@@ -786,7 +786,7 @@ def _search(
         def default_doc_builder(hit: Dict) -> Document:
             return Document(
                 page_content=hit["_source"].get(self.query_field, ""),
-                metadata=hit["_source"]["metadata"],
+                metadata=hit["_source"]["metadata"] if "metadata" in hit["_source"] else {},
```
Comment (joemcelroy) [inline libs/langchain/langchain/vectorstores/elasticsearch.py:789]:
```suggestion
                metadata=hit["_source"].get("metadata", {}),
```

Could you also add tests for this please?

**Comment by joemcelroy** [comment]
hi there @RodolpheCalvet - Thanks for your contribution, this is a good addition :)

Yes you're fine to update the existing PR.

**Comment by baskaryan** [comment]
elasticsearch integration has been moved to it's own package and repo, we should re-open this fix against langchain-elastic repo here: https://github.com/langchain-ai/langchain-elastic/blob/main/libs/elasticsearch/langchain_elasticsearch/vectorstores.py

**Pull request closed without merging**